src/util/RoundRobinLB.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AbstractLoadBalancer} from './AbstractLoadBalancer';
+import {randomInt} from '../Util';
+import {Member} from '../core/Member';
+
+/**
+ * A {@link LoadBalancer} implementation that relies on using round robin
+ * to a next member to send a request to.
+ */
+export class RoundRobinLB extends AbstractLoadBalancer {
+    private index: number;
+
+    constructor() {
+        super();
+        this.index = randomInt(Date.now());
+    }
+
+    next(): Member {
+        const members = this.getMembers();
+        if (members == null || members.length === 0) {
+            return null;
+        }
+
+        const length = members.length;
+        const idx = (this.index++) % length;
```

Comment:
Java client has overflow protection for `this.index`. See the extra `(<base calculation> + length) % length` part here:
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/client/util/RoundRobinLB.java#L49

On the other hand, we don't have `int` type in JS and `Number.MAX_SAFE_INTEGER` is very large (see #535), so we should be safe when it comes to the overflow. Yet, it still makes sense to cap the initial seed value for index somehow in order to make sure that the initial value is not very big. WDYT?

Reply:
I set the cap as `ceil(Number.MAX_SAFE_INTEGER / 1024)`. It would take 285 years to reach `Number.MAX_SAFE_INTEGER` even if we call `next` 1 million times per second so I think it is okay to do so. By the way, if the seed is somehow greater than this, I assigned the seed to a random integer between 0 and this limit. Is it okay ?

Reply:
What was the reason to choose `Math.ceil(Number.MAX_SAFE_INTEGER / 1024)`? Something like `1 << 16` sound like a more reasonable cap. After all, the cap should be close enough to cluster sizes in the real world.

Reply:
I'd also propose to simplify the constructor to:
```js
    constructor() {
        super();
        this.index = randomInt(INITIAL_SEED_CAP);
    }
```

The reason is that `process.hrtime()[1]` sooner or later will become larger than the cap and we can simply use one source of randomization. WDYT?

Reply:
I think this is better than the nanotime based solution.